### PR TITLE
V3 pension field calculator

### PR DESCRIPTION
**Pull request by anemne**
## Short Description

Added pension field with validation for showCalculator. Created as own object if we later want to add some extra functionality to it. 

---

## Visual Overview (Image/Video)

https://github.com/user-attachments/assets/28ddb644-af3f-48a3-b7fb-de1bf276d2bc

---

## Checklist

Please ensure that you’ve completed the following checkpoints before submitting your pull request:

- [x] **Documentation**: Relevant documentation has been added or updated (if applicable).
- [x] **Testing**: Have you tested your changes thoroughly?
  - Please list the types of tests you've run (unit, integration, manual, etc.):

### CODE REVIEW

File: studio/schemas/documents/compensations.ts [hunk 1]
```diff
@@ -19,7 +20,14 @@ const compensations = defineType({
         "Enter the primary title that will be displayed at the top of the compensation page. This is what users will see when they visit the page.",
     },
     titleSlug,
-    // add pension here. pension doesn't rely on locations
+    defineField({
+      name: "showSalaryCalculator",
+      title: "Show Salary Calculator",
+      description: "Should the salary calculator be visible on the page?",
```
Comment (christinaroise) [inline studio/schemas/documents/compensations.ts:26]:
```suggestion
      description: "Toggle this setting to show or hide the salary calculator on the page.",
```

Should we maybe add some explanation as to what that calculator is?? Like that it's a design feature that calculates the estimated salary? 🤷‍♀️

File: studio/schemas/objects/compensations/pension.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { defineField } from "sanity";
+
+export const pension = defineField({
+  name: "pensionData",
+  title: "Pension Percentage",
+  type: "number",
+  initialValue: 7,
+  description:
+    "Enter the percentage of the pension offered by Variant when employed. The value should be a positive number and will be used to calculate the pension amount.",
```
Comment (christinaroise) [inline studio/schemas/objects/compensations/pension.ts:9]:
```suggestion
    `Specify the percentage of the pension provided by Variant for employees. The value should be a positive number and will be used to calculate the pension amount.`,
```

File: studio/schemas/objects/compensations/pension.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  validation: (Rule) =>
+    Rule.custom((value, context) => {
+      if (
+        context.document?.showSalaryCalculator &&
+        (value === undefined || value === null)
+      ) {
+        return "Pension percentage is required when the salary calculator is shown.";
```
Comment (christinaroise) [inline studio/schemas/objects/compensations/pension.ts:16]:
```suggestion
        return "Please enter a pension percentage. This is required when the salary calculator is enabled.";
```

File: studio/schemas/objects/compensations/pension.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      }
+      if (value !== undefined && (value < 0 || value > 100)) {
+        return "Pension percentage must be a positive value and less than or equal to 100";
```
Comment (christinaroise) [inline studio/schemas/objects/compensations/pension.ts:19]:
```suggestion
        return "The pension percentage must be a number between 0 and 100.";
```

**Review by christinaroise — APPROVED** [review]
Awesome. I only have some language suggestions for a more user friendly approach

File: studio/schemas/objects/compensations/pension.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        context.document?.showSalaryCalculator &&
+        (value === undefined || value === null)
+      ) {
+        return "Please enter a pension percentage. This is required when the salary calculator is enabled.";
+      }
+      if (value !== undefined && (value < 0 || value > 100)) {
+        return "The pension percentage must be a number between 0 and 100.";
+      }
+      return true;
+    }),
```
Comment (mathiazom) [inline studio/schemas/objects/compensations/pension.ts:21]:
```suggestion
  validation: (Rule) => [
    Rule.min(0)
      .max(100)
      .error("The pension percentage must be a number between 0 and 100."),
    Rule.custom((value, context) => {
      if (
        context.document?.showSalaryCalculator &&
        (value === undefined || value === null)
      ) {
        return "Please enter a pension percentage. This is required when the salary calculator is enabled.";
      }
      return true;
    }),
  ],
```

**Review by mathiazom — APPROVED** [review]
Looks good, suggested a small validation refactor if you want it 🤗

**Pull request merged**